controllers/user/UserController.php

```diff
@@ -98,17 +107,92 @@ public function actionCreate(){
     }
 
     /**
-     * Authenticate the user by username and password
-     * If, the credentials are correct then token will be generated by JWT
+     * Store the user details like email, token in user_invite table
+     * Generate six digit random number code
+     * @param $email
+     * @return array|false
+     * @throws \Throwable
+     * @throws \yii\db\StaleObjectException
+     */
+    public function inviteUser($email){
+        $invite = new Invite();
+        $invite->scenario = 'invite';
+
+        //If the site in maintenance mode
+        if (Yii::$app->settings->get('maintenanceMode')) {
+            Yii::$app->getView()->warn(ControllerAccess::getMaintenanceModeWarningText());
+            return false;
+        }
+
+        $invite->source = Invite::SOURCE_SELF;
+        $invite->language = Yii::$app->language;
+        $invite->email = $email;
+
+        // Delete existing invite for e-mail - but reuse token
+        $existingInvite = Invite::findOne(['email' => $email]);
+        if ($existingInvite !== null) {
+            $invite->token = $existingInvite->token;
+            $existingInvite->delete();
+        }
+
+        $invite->token = $this->generateCode();
+        $checkCodeAlredyExist = Invite::find()->where(['token'=>$invite->token])->one();
+
+        if($checkCodeAlredyExist!=null){
+            $invite->token = $this->generateCode();
+        }
+
+        //We set validation false because we are not fill recaptcha value in API
+        if($invite->save(false)){
+            $this->sendMail($invite);
+            return $this->returnSuccess("Registration successfully done, Please check your email, Code is send to your email address ".$email,200);
+        }
+        return $this->returnError(400,$invite->errors);
+
+    }
+
+    public function generateCode()
+    {
+        return random_int(100000, 999999);
```

Comment:
would a 6 digit `uuid`/`uniqid` be more "safe" here? a random number between 100000 and 999999 would work in most cases, but there is a small chance to have same codes at a certain time for some users, right?

Reply:
all right. for the expected amount of users this should work fine. thanks.